diff --git a/lz4/lib/tests/fuzzer.c b/lz4/lib/tests/fuzzer.c
index b129c96..1703ddc 100644
--- a/lz4/lib/tests/fuzzer.c
+++ b/lz4/lib/tests/fuzzer.c
@@ -656,15 +656,16 @@ _output_error:
 #define testCompressedSize (128 KB)
 #define ringBufferSize (8 KB)
 
+char testInput[testInputSize];
+char testCompressed[testCompressedSize];
+char testVerify[testInputSize];
+char ringBuffer[ringBufferSize];
+
 static void FUZ_unitTests(int compressionLevel)
 {
     const unsigned testNb = 0;
     const unsigned seed   = 0;
     const unsigned cycleNb= 0;
-    char testInput[testInputSize];
-    char testCompressed[testCompressedSize];
-    char testVerify[testInputSize];
-    char ringBuffer[ringBufferSize];
     U32 randState = 1;
 
     /* Init */
