diff --git a/JniConstants.cpp b/JniConstants.cpp
index cfea131..4b5dea7 100644
--- a/JniConstants.cpp
+++ b/JniConstants.cpp
@@ -26,7 +26,9 @@
 #include <mutex>
 
 static std::atomic<bool> g_constants_initialized(false);
+#if 0
 static std::mutex g_constants_mutex;
+#endif
 
 jclass JniConstants::booleanClass;
 jclass JniConstants::byteArrayClass;
@@ -87,11 +89,13 @@ void JniConstants::init(JNIEnv* env) {
     }
 
     // Slightly slower check
+#if 0
     std::lock_guard<std::mutex> guard(g_constants_mutex);
     if (g_constants_initialized) {
       // already initialized
       return;
     }
+#endif
 
     booleanClass = findClass(env, "java/lang/Boolean");
     byteArrayClass = findClass(env, "[B");
