diff --git a/libnativebridge/libnativebridge/tests/CodeCacheStatFail_test.cpp b/libnativebridge/libnativebridge/tests/CodeCacheStatFail_test.cpp
index 4ea519ee9..e3594c496 100644
--- a/libnativebridge/libnativebridge/tests/CodeCacheStatFail_test.cpp
+++ b/libnativebridge/libnativebridge/tests/CodeCacheStatFail_test.cpp
@@ -32,7 +32,7 @@ TEST_F(NativeBridgeTest, CodeCacheStatFail) {
 
     struct stat st;
     ASSERT_EQ(-1, stat(kCodeCacheStatFail, &st));
-    ASSERT_EQ(ENOTDIR, errno);
+    ASSERT_EQ(ENOENT, errno);
 
     // Init
     ASSERT_TRUE(LoadNativeBridge(kNativeBridgeLibrary, nullptr));
