diff --git a/libnativebridge/libnativebridge/tests/NativeBridge2Signal_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge2Signal_test.cpp
index 44e45e362..f79e239d2 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge2Signal_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge2Signal_test.cpp
@@ -21,7 +21,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary2 = "libnativebridge2-dummy.so";
+constexpr const char* kNativeBridgeLibrary2 = "libnativebridge2-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V2_Signal) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3CreateNamespace_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3CreateNamespace_test.cpp
index 668d9428a..dd32407b3 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3CreateNamespace_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3CreateNamespace_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_CreateNamespace) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3GetError_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3GetError_test.cpp
index 0b9f5821f..583f0185b 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3GetError_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3GetError_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_GetError) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3InitAnonymousNamespace_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3InitAnonymousNamespace_test.cpp
index b0d6b09d1..87b9e166c 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3InitAnonymousNamespace_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3InitAnonymousNamespace_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_InitAnonymousNamespace) {
   // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3IsPathSupported_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3IsPathSupported_test.cpp
index 325e40b40..10b864864 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3IsPathSupported_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3IsPathSupported_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_IsPathSupported) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3LoadLibraryExt_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3LoadLibraryExt_test.cpp
index 4caeb4481..cee7ae70a 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3LoadLibraryExt_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3LoadLibraryExt_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_LoadLibraryExt) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridge3UnloadLibrary_test.cpp b/libnativebridge/libnativebridge/tests/NativeBridge3UnloadLibrary_test.cpp
index 93a979c0a..fc5afaa84 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridge3UnloadLibrary_test.cpp
+++ b/libnativebridge/libnativebridge/tests/NativeBridge3UnloadLibrary_test.cpp
@@ -18,7 +18,7 @@
 
 namespace android {
 
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 TEST_F(NativeBridgeTest, V3_UnloadLibrary) {
     // Init
diff --git a/libnativebridge/libnativebridge/tests/NativeBridgeTest.h b/libnativebridge/libnativebridge/tests/NativeBridgeTest.h
index 0f99816c9..d8256aefa 100644
--- a/libnativebridge/libnativebridge/tests/NativeBridgeTest.h
+++ b/libnativebridge/libnativebridge/tests/NativeBridgeTest.h
@@ -22,11 +22,11 @@
 #include <nativebridge/native_bridge.h>
 #include <gtest/gtest.h>
 
-constexpr const char* kNativeBridgeLibrary = "libnativebridge-dummy.so";
+constexpr const char* kNativeBridgeLibrary = "libnativebridge-dummy.lib.so";
 constexpr const char* kCodeCache = "./code_cache";
 constexpr const char* kCodeCacheStatFail = "./code_cache/temp";
-constexpr const char* kNativeBridgeLibrary2 = "libnativebridge2-dummy.so";
-constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.so";
+constexpr const char* kNativeBridgeLibrary2 = "libnativebridge2-dummy.lib.so";
+constexpr const char* kNativeBridgeLibrary3 = "libnativebridge3-dummy.lib.so";
 
 namespace android {
 
