diff --git a/system/core/base/include/android-base/chrono_utils.h b/system/core/base/include/android-base/chrono_utils.h
index 7679d4c94..d35079d59 100644
--- a/system/core/base/include/android-base/chrono_utils.h
+++ b/system/core/base/include/android-base/chrono_utils.h
@@ -20,8 +20,6 @@
 #include <chrono>
 #include <sstream>
 
-using namespace std::chrono_literals;
-
 namespace android {
 namespace base {
 
diff --git a/system/core/base/properties.cpp b/system/core/base/properties.cpp
index 816bca0eb..ca17c4e25 100644
--- a/system/core/base/properties.cpp
+++ b/system/core/base/properties.cpp
@@ -27,8 +27,6 @@
 
 #include <android-base/parseint.h>
 
-using namespace std::chrono_literals;
-
 namespace android {
 namespace base {
 
diff --git a/system/core/base/properties_test.cpp b/system/core/base/properties_test.cpp
index de5f3dcfa..d4eb6f116 100644
--- a/system/core/base/properties_test.cpp
+++ b/system/core/base/properties_test.cpp
@@ -23,8 +23,6 @@
 #include <string>
 #include <thread>
 
-using namespace std::chrono_literals;
-
 TEST(properties, smoke) {
   android::base::SetProperty("debug.libbase.property_test", "hello");
 
