diff --git a/system/core/base/logging_test.cpp b/system/core/base/logging_test.cpp
index adb041b32..bf69ee9dd 100644
--- a/system/core/base/logging_test.cpp
+++ b/system/core/base/logging_test.cpp
@@ -595,6 +600,6 @@ TEST(logging, LOG_FATAL_ABORTER_MESSAGE) {
   EXPECT_EQ(CountLineAborter::newline_count, 1U + 1U);  // +1 for final '\n'.
 }
 
-__attribute__((constructor)) void TestLoggingInConstructor() {
-  LOG(ERROR) << "foobar";
-}
+//__attribute__((constructor)) void TestLoggingInConstructor() {
+//  LOG(ERROR) << "foobar";
+//}
