diff --git a/system/core/base/file_test.cpp b/system/core/base/file_test.cpp
index 02b431de7..7da1cc3b7 100644
--- a/system/core/base/file_test.cpp
+++ b/system/core/base/file_test.cpp
@@ -182,7 +182,7 @@ TEST(file, Realpath) {
 
   {
     errno = 0;
-    std::string path = std::string(td.path) + "/foo.noent";
+    std::string path = std::string(td.path) + "/foo.noent/";
     std::string result = "wrong";
     ASSERT_TRUE(!android::base::Realpath(path, &result));
     ASSERT_TRUE(result.empty());
