diff --git a/system/core/liblog/include/log/log_time.h b/system/core/liblog/include/log/log_time.h
index 3764faf75..40d607fa1 100644
--- a/system/core/liblog/include/log/log_time.h
+++ b/system/core/liblog/include/log/log_time.h
@@ -56,7 +56,7 @@ struct log_time {
 #endif
   log_time() {
   }
-#ifdef __linux__
+#if defined(__linux__) || defined(__FreeBSD__)
   explicit log_time(clockid_t id) {
     timespec T;
     clock_gettime(id, &T);
